ops/terraform/services/server/modules/bfd_server_error_alerts/lambda_src/bfd_server_errors_alerter_scheduler.py

```diff
@@ -0,0 +1,246 @@
+            GroupName=EVENTBRIDGE_SCHEDULES_GROUP,
+            Name=schedule.name,
+            ScheduleExpression=schedule.expression,
+            ScheduleExpressionTimezone="UTC",
+            Target={"RoleArn": SCHEDULER_ROLE_ARN, "Arn": ALERTER_LAMBDA_ARN},
+            FlexibleTimeWindow={"Mode": "OFF"},
+        )
+    except scheduler_client.exceptions.ClientError:
+        logger.error(
+            "An unrecoverable error occurred when trying to schedule %s: ",
+            schedule.name,
+            exc_info=True,
+        )
+        return False
+
+    logger.info("Schedule %s created successfully", schedule.name)
+    return True
+
+
+def handler(event: Any, context: Any):
+    if not all(
+        [
+            REGION,
+            BFD_ENVIRONMENT,
+            EVENTBRIDGE_SCHEDULES_GROUP,
+            RECURRING_SCHEDULE_RATE_STR,
+        ]
+    ):
+        logger.error("Not all necessary environment variables were defined, exiting...")
+        return
+
+    try:
+        record: dict[str, Any] = event["Records"][0]
+    except KeyError as ex:
+        logger.error("The incoming event was invalid: %s", repr(ex))
+        return
+    except IndexError:
+        logger.error("Invalid event notification, no records found")
+        return
+
+    try:
+        sns_message_json: str = record["Sns"]["Message"]
+        alarm_details = json.loads(sns_message_json)
+        alarm_state = AlarmState(alarm_details["NewStateValue"])
+    except KeyError:
+        logger.error("No message found in SNS notification")
+        return
+    except json.JSONDecodeError:
+        logger.error("SNS message body was not valid JSON")
+        return
+
+    # Delete old OnetimeSchedules by listing all OnetimeSchedules within the group and deleting
+    # those with names indicating times that have since passed
+    try:
+        all_onetime_schedules = [
+            one_time_schedule
+            for schedule in scheduler_client.list_schedules(
+                GroupName=EVENTBRIDGE_SCHEDULES_GROUP,
+                NamePrefix=ONETIME_SCHEDULE_NAME_PREFIX,
+            )["Schedules"]
+            if "Name" in schedule
+            and ((one_time_schedule := OnetimeSchedule.from_name(schedule["Name"])) is not None)
+        ]
+        all_schedules_to_delete = [x for x in all_onetime_schedules if x.time <= datetime.utcnow()]
+        for schedule in all_schedules_to_delete:
+            logger.info(
+                "Deleting schedule %s as its invocation time has passed...",
+                schedule.name,
+            )
+            scheduler_client.delete_schedule(
+                GroupName=EVENTBRIDGE_SCHEDULES_GROUP, Name=schedule.name
+            )
+            logger.info("Schedule %s deleted successfully", schedule.name)
+    except scheduler_client.exceptions.ClientError:
+        logger.error(
+            "An error occurred when trying to delete old OnetimeSchedules; continuing. Err: ",
+            exc_info=True,
+        )
```

Comment:
One-time EventBridge Schedules [do not automatically remove themselves](https://docs.aws.amazon.com/scheduler/latest/UserGuide/managing-schedule-delete.html) after executing, so before doing any work the Scheduler will remove expired one-time schedules from the corresponding Schedule Group. This way, at most we'll only ever have a single, expired one-time schedule in each environment's Schedule Group.

Reply:
Thanks for the explanation!